xmppserver/src/main/java/org/jivesoftware/openfire/muc/MUCRole.java

```diff
@@ -444,4 +674,137 @@ public static Affiliation valueOf(int value) {
             }
         }
     }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((nick == null) ? 0 : nick.hashCode());
+        result = prime * result + ((rJID == null) ? 0 : rJID.hashCode());
+        result = prime * result + ((roomJid == null) ? 0 : roomJid.hashCode());
+        result = prime * result + ((userJid == null) ? 0 : userJid.hashCode());
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj)
+            return true;
+        if (obj == null)
+            return false;
+        if (getClass() != obj.getClass())
+            return false;
+        MUCRole other = (MUCRole) obj;
+        if (nick == null) {
```

Comment:
Similarly, perhaps `Objects.equal(this.nick, other.nick) && Objects.equal (this.rJID && other.rJID) ...`

Reply:
Again, generated code. Can't be wrong. ;)